src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java

```diff
@@ -123,18 +126,29 @@ public Page<AdvertisementTitleDto> findAdvertisementsHavingCategory(
             @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size) throws CategoryIdNotFoundException {
         if (!categoryService.isCategoryExistsById(categoryId)) {
             throw new CategoryIdNotFoundException(
-                    getParametrizedMessageSource(ResponseMessagesHandler.ValidationMessage.INVALID_CATEGORY_ID, categoryId));
+                    getParametrizedMessageSource(INVALID_CATEGORY_ID, categoryId));
         }
         return advertisementService.findByCategoryId(categoryId, PageRequest.of(page, size));
     }
 
-    @PostMapping(value = ApiKey.ADV_FILTER, produces = MediaType.APPLICATION_JSON_VALUE)
-    @Operation(summary = "Filter advertisements by multiple params and return up to 10 results.\n" +
-            "Fill only needed parameters.")
+    @GetMapping(value = ApiKey.ADV_FILTER, produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Filter advertisements by multiple params")
     @ApiResponse(responseCode = "200", description = "OK")
     @ResponseStatus(HttpStatus.OK)
-    public List<AdvertisementTitleDto> getFirst10BySearchParameters(@Valid @RequestBody AdvertisementFilterDto filterDto) {
-        return advertisementService.findFirst10ByFilter(filterDto);
+    public Page<AdvertisementTitleDto> findAdvertisementBySearchParameters(@Valid @ParameterObject AdvertisementFilterRequest filterDto) throws CategoryIdNotFoundException {
+        validateCategoryCombinationSize(filterDto.getSubcategorySearchRequest().getCategoryId(), filterDto.getAdvertisementFilter());
+        return advertisementService.findAdvertisementByFilter(filterDto);
+    }
+
+    private void validateCategoryCombinationSize(long categoryId, AdvertisementFilter filter) throws CategoryIdNotFoundException {
+        String categoryName = categoryService.findCategoryDtoById(categoryId)
```

Comment:
You can simply check if all the subcategory ID belongs to provided category withing a single SQL request.